docs/proposals/data-cache.md

```diff
@@ -0,0 +1,160 @@
+
+## Implementation
+
+### Sequence Diagram
+
+FYR. The final implementation may be slightly different from this diagram.
+
+![](../img/data_cache_sequence.png)
+The general process is as follows:
+
+1. The user submits a yaml file describing the job whose field contains the parameters of the data cache, which means that the job will use the feature of the data cache.
+2. KubeDL's Job Controller detects that a job with data cache parameters has been created, and it passes the parameters to the handler responsible for enable data cache.
+3. After verifying that the parameters are correct, the cache handler adds defaults to the missing parameters and creates a client. The client checks to see if Fluid has been deployed in the Kubernetes cluster, and if so, it will create the dataset and cache runtime by the api server.
+4. The volume controller from Fluid detects that there is a pair of dataset and runtime in the cluster that can be bound, and it will automatically create pv and assign pvc to it. Users can enjoy data cache by mounting this pvc.
+5. When the listener in KubeDL detects that pvc is created, it will notify the cache handler and pass relevant information about pvc to the handler.
+6. Handler receives the parameters of the pvc and injects the pvc into the container of the KubeDL job.
+
+With the above process, the dataset in the specified file system can be cached, which means that the time for the KubeDL job to access the dataset will be greatly reduced.
+
+### Cache Configuration
+
+Reference to the configuration of `modelVersion` in `example/tf/tf_job_mnist_modelversion.yaml`
+
+```yaml
+    apiVersion: "training.kubedl.io/v1alpha1"
+    kind: "TFJob"
+    metadata:
+      name: "mnist"
+      namespace: kubedl
+    spec:
+      cleanPodPolicy: None
++     fluidConfig:
++ 			runtimeType: "AlluxioRuntime"
++     	levels:
++     		- mountPoint: "/file/system/dataset/path"
++     			mountPath: "/path/in/container"
++     			quota: "4Gi"
+      tfReplicaSpecs:
+        ...
+        No additional volumes need to be configured. 
+        The cache handler will automatically inject the pvc created by fluid into the container
+```
+
+Compared with the first configuration method mentioned in **Alternatives Considered**, the advantage of this configuration data cache is that:
+1. The structure of the above method is clearer and easier to parse or use than using plain strings to configure parameters
+2. This method is more conducive to supporting the hierarchical storage configuration provided by Fluid, while it is difficult to describe the multilayer and multidirectory configurations using plain strings
+
+A partial list of supported options, full list TBD
+
+```yaml
+# Required parameters for dataset
+mountPoint: "https://mirrors.bit.edu.cn/apache/spark/"   # Directory of the dataset to be mounted(required)
```

Comment:
Does mountPoint mean the data source, the name 'mountPoint' is a bit confusing I feel.
maybe just dataSource ? or dataSet